main.cpp

```diff
@@ -238,153 +353,306 @@ struct Bird
     float clockSpeed = 3.5f;
-    //    4) cores (int)
     int cores = 6;
-    //    5) socket size (std::string)
     std::string socket = "AM4";
-    //3 things it can do:
-    //    1) store data
-    float storeData(); //returns stored values
-    //    2) output results
-    float outputResults(); //returns current values
-    //    3) compute
-    float compute(float x, float y);//returns result of operations on inputs
+
+    float storeData(); 
+    float outputResults(); 
+    float compute(float x, float y);
 };
 
+float CPU::storeData()
+{
+    float data = 1.43667f * 2.44435f * temperature;
+    return data;
+}
+
+float CPU::outputResults()
+{
+    float data = 89.667f * .30741f * temperature;
+    return data;
+}
+
+float CPU::compute(float x, float y)
+{
+    float data = x * y * temperature;
+    return data;
+}
+
 struct Motherboard
 {
-    //5 properties:
-    //    1) manufacturer (std::string)
     std::string manufacturer = "Gigabyte";
-    //    2) wireless LAN capability (bool)
     bool wirelessLAN = false;
-    //    3) number of m.2 slots (int)
     int m2Slots = 2;
-    //    4) number of expansion slots (int
     int expansionSlots = 4;
-    //    5) max RAM capacity (int)
     int maxRAM = 64;
-    //3 things it can do:
-    //    1) transmit data to RAM
+    int AVSignals = 0;
+    
     void dataToRAM();
-    //    2) generate A/V signals
     void generateAVSignals();
-    //    3) auto suspend
     void autoSuspend();
 };
 
+void Motherboard::dataToRAM()
+{
+    maxRAM++;
+}
+
+void Motherboard::generateAVSignals()
+{
+    AVSignals++;
+}
+
+void Motherboard::autoSuspend()
+{
+    maxRAM = 0;
+    AVSignals = 0;
+}
+
 struct KeyboardAndMouse
 {
-    //5 properties:
-    //    1) number of mouse buttons (int)
     int numberOfMouseButtons = 3;
-    //    2) keyboard style chiclet/mechanical (bool)
     bool mechanicalStyle = false;
-    //    3) wireless yes/no (bool)
     bool wireless = true;
-    //    4) language setting (std::string)
     std::string language = "English";
-    //    5) numpad yes/no (bool)
     bool numpad = false;
-    //3 things it can do:
-    //    1) transmit keystroke data
-    int transmitKeystrokeData(); //returns ASCII value
-    //    2) transmit mouse XY change
-    float transmitXYChange(); //returns distance from a 0,0 position, which refreshes periodically
-    //    3) toggle status light (like capslock)
-    bool toggleStatus(std::string status);//in which indicator to toggle, out isLit true or false
+
+    int transmitKeystrokeData();
+    double transmitXYChange(double x, double y);
+    bool toggleStatus(bool status);
 };
 
+int KeyboardAndMouse::transmitKeystrokeData()
+{
+    return 103;
+}
+
+double KeyboardAndMouse::transmitXYChange(double x, double y)
+{
+    return sqrt(pow(x, 2.) + pow(y, 2.));
+}
+
+bool toggleStatus(bool status)
+{
+    if(status == true)
+    {
+        return false;
+    }
+    else
+    {
+        return true;
+    }
+}
```

Comment:
This is not a member function
- [ ] revise
- [ ] revise: if/else return statements